addon/-private/build-spec-processor.js

```diff
@@ -0,0 +1,26 @@
+import Binding from './binding';
+import ComponentSpec from './component-spec';
+import { transform, rule, simple, rest } from 'botanist';
+
+export default function buildSpecProcessor({ bindKey, componentKey }) {
+  return transform({
+    @rule({ [bindKey]: simple('name') })
+    createBinding({ name }) {
+      if (!name) {
+        throw new Error(`Invalid binding: ""`);
+      } else {
+        return new Binding(name);
+      }
+    },
+
+    @rule({ [componentKey]: simple('name'), ...rest('config') })
+    createComponentSpec({ name, config }, { resolveComponent, owner }) {
+      const path = resolveComponent(name);
+      if (!owner.hasRegistration(`component:${path}`)) {
```

Comment:
Unless you meant the `{{exclaim-ui}}` component instance, in which case just passing the owner feels better to me from a Law of Demeter perspective